geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionAdvisor.java

```diff
@@ -1680,4 +1544,262 @@ public void fillInToString(StringBuilder sb) {
       return null;
     }
   }
+
+
+
+  private static class OperationMonitor {
```

Comment:
I think it would be helpful if this class was in its own file. As it is now it has a number of "private" methods that are part of its external contract. At the least, I think it would help if the methods on this class that are called by other classes were made "public".